jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/info/ThreadGroupInfo.java

```diff
@@ -0,0 +1,220 @@
+/*[INCLUDE-IF Sidecar18-SE]*/
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package openj9.tools.attach.diagnostics.info;
+
+import static com.ibm.oti.util.Msg.getString;
+import java.io.IOException;
+import java.lang.management.LockInfo;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.util.Properties;
+
+import com.ibm.tools.attach.target.IPC;
+import openj9.tools.attach.diagnostics.base.DiagnosticProperties;
+import openj9.tools.attach.diagnostics.base.DiagnosticsInfo;
+
+/**
+ * Class for encoding and decoding stack traces for transmission via attach API.
+ * This mimics the behaviour of Thread
+ */
+public class ThreadGroupInfo implements DiagnosticsInfo {
+
+	private static final String CURRENT_GROUPINFO_VERSION = "0"; //$NON-NLS-1$
+	private static final String TIME_FORMAT_STRING = "\t%s time: %d ns%n"; //$NON-NLS-1$
+	static final String THREAD_GROUP_PREFIX = "threadgroup."; //$NON-NLS-1$
+	static final String CPU_TIME = "cpu_time"; //$NON-NLS-1$
+	static final String USER_TIME = "user_time"; //$NON-NLS-1$
+	private static final String THREAD_GROUP_VERSION = THREAD_GROUP_PREFIX + "version"; //$NON-NLS-1$
+	private static final String NUM_THREADS = THREAD_GROUP_PREFIX+"num_threads"; //$NON-NLS-1$
+	private static ThreadMXBean threadBean;
```

Comment:
This is not thread-safe: it should be passed to static methods that need it.

Reply:
Good plan.